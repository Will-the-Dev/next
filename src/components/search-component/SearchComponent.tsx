import React, { ChangeEvent, FC, FormEvent } from 'react';
import { Button, TextField, Typography } from '@material-ui/core';
import styled from 'styled-components';

const FindContainer = styled.div`
    margin: 30px auto;
    max-width: 570px;
    z-index: 1;
`;

const FlexRowContainer = styled.div`
    display: flex;
    justify-content: center;

    div:first-child {
        margin-right: 15px;
    }
`;

type Props = {
    onChangeText: (e: ChangeEvent<HTMLInputElement>) => void;
    onSubmitForm: (e: FormEvent<HTMLFormElement>) => void;
};

export const SearchComponent: FC<Props> = ({ onSubmitForm, onChangeText }) => (
    <FindContainer>
        <form data-testid="form" onSubmit={onSubmitForm}>
            <Typography align="left" color="inherit" variant="h4">
                Search for song, artist or album.
            </Typography>
            <br />
            <FlexRowContainer>
                <TextField
                    type="search"
                    InputLabelProps={{
                        shrink: true,
                    }}
                    color="secondary"
                    onChange={onChangeText}
                    placeholder="search"
                    variant="outlined"
                    fullWidth
                />
                <Button variant="contained" type="submit">
                    Submit
                </Button>
            </FlexRowContainer>
        </form>
    </FindContainer>
);
