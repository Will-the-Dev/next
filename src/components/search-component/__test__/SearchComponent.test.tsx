import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import { SearchComponent } from '../SearchComponent';

describe('SearchComponent tests', () => {
    it('should render the component to match the snapshot correctly', () => {
        const { asFragment } = render(
            <SearchComponent onChangeText={() => {}} onSubmitForm={() => {}} />
        );

        expect(asFragment()).toMatchSnapshot();
    });

    it('should call the submit form callback prop when form is submitted', () => {
        const onSubmitMock = jest.fn();
        const { getByTestId } = render(
            <SearchComponent
                onChangeText={() => {}}
                onSubmitForm={onSubmitMock}
            />
        );

        fireEvent.submit(getByTestId('form'));
        expect(onSubmitMock).toHaveBeenCalled();
    });
});
