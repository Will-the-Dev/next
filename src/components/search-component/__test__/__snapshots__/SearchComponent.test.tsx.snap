// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchComponent tests should render the component to match the snapshot correctly 1`] = `
<DocumentFragment>
  .c0 {
  margin: 30px auto;
  max-width: 570px;
  z-index: 1;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 div:first-child {
  margin-right: 15px;
}

<div
    class="c0"
  >
    <form
      data-testid="form"
    >
      <h4
        class="MuiTypography-root MuiTypography-h4 MuiTypography-colorInherit MuiTypography-alignLeft"
      >
        Search for song, artist or album.
      </h4>
      <br />
      <div
        class="c1"
      >
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorSecondary MuiOutlinedInput-colorSecondary MuiInputBase-fullWidth MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputTypeSearch"
              placeholder="search"
              type="search"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
              style="padding-left: 8px;"
            >
              <legend
                class="PrivateNotchedOutline-legend-2"
                style="width: 0.01px;"
              >
                <span>
                  â€‹
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained"
          tabindex="0"
          type="submit"
        >
          <span
            class="MuiButton-label"
          >
            Submit
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
